### PR TITLE
Added FailedTestRemover tool.

### PR DESCRIPTION
**Pull request by ReedOei**

### CODE REVIEW

**Comment by ReedOei** [comment]
Tool does not comment out, just removes. I encountered a weird bug where it would add the comments to the object representing the file, but would not write them out to the file with the rest of the files content when running on the VM, but would work when on my own computer. 
I can try to fix it if it's an important feature.

File: dt-impact-tracer/src/edu/washington/cs/dt/impact/tools/FailedTestRemover.java [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+         */
+        private boolean removeMethod(final MethodDeclaration method) {
+            for (final ClassOrInterfaceDeclaration classDeclaration : classList) {
+                boolean success = classDeclaration.getMembers().remove(method);
+
+                // Doesn't seem to work on VM for some reason.
+                // classDeclaration.addOrphanComment(new BlockComment(method.toString()));
+
+                if (!success) {
+                    return false;
```
Comment (winglam) [inline dt-impact-tracer/src/edu/washington/cs/dt/impact/tools/FailedTestRemover.java]: If the goal of this method is indeed to "remove the method declaration from all classes" then you should not return false here. You can use a boolean flag to keep track of whether you were successful in removing this method from at least one class and return the flag at the end of the method.

File: dt-impact-tracer/src/edu/washington/cs/dt/impact/tools/FailedTestRemover.java [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    }
+
+    private FailedTestRemover(final String filename, final String classPath, final String outfileName)
+            throws IOException, ParseException {
+        javaFile = new JavaFile(filename, classPath, outfileName);
+    }
+
+    public static void main(final String[] args) throws Exception {
+        if (args.length < 2) {
+            System.out.println("Usage: java FailedTestRemover <classpath> <filename> [<output-dir>]");
```
Comment (winglam) [inline dt-impact-tracer/src/edu/washington/cs/dt/impact/tools/FailedTestRemover.java]: Please expand the help message here with a description of what the class is meant to do, what the arguments are meant for and an example of using this class with one of our subjects.

File: experiments/generate-sootTestOutput-auto.sh [hunk 3]
```diff
@@ -20,16 +20,29 @@ else
    javac -cp $DT_LIBS:$DT_CLASS:$DT_TOOLS: RegressionTest*.java
 fi
 
-mkdir bin
+mkdir -p bin
 mv *.class ./bin
 cd $DT_SUBJ
 
-#check if auto tests are can be compiled with new subject
+# 7. Move auto tests to new subject/comment out those that don't compile
```
Comment (winglam) [inline experiments/generate-sootTestOutput-auto.sh]: Please move step 7 to come after step 6 in this file.

Comment (winglam) [reply experiments/generate-sootTestOutput-auto.sh]: To fully complete step 7, we would have to copy the Randoop tests to the new subject, remove all the ones that do not compile, and finally generate a $NEW_DT_SUBJ/$SUBJ_NAME-auto-order file. This file will contain the list of Randoop tests that the new subject is actually compatible with and you can get that list by doing something similar to the existing step 4 of "Instructions to generate automatically-generated tests for a subject".

Your changes so far accomplish everything but generating a $NEW_DT_SUBJ/$SUBJ_NAME-auto-order file. Please amend your changes to this file to include this step.

Comment (winglam) [reply experiments/generate-sootTestOutput-auto.sh]: Once your changes to this file are complete, please take the relevant code segments from this file and update the Google sites guidelines for step 7 in "Instructions to generate automatically-generated tests for a subject" to reflect your awesome work to automate this step!

File: dt-impact-tracer/src/edu/washington/cs/dt/impact/tools/FailedTestRemover.java (hunk 1 above)
Comment (winglam) [reply dt-impact-tracer/src/edu/washington/cs/dt/impact/tools/FailedTestRemover.java]: For each method that you end up removing, please record the fully qualified name of that method (https://stackoverflow.com/questions/11778032/fully-qualified-reference-to-java-method-signature) and save all such tests into a file called $SUBJ_NAME-auto-removed-tests in $NEW_DT_SUBJ. 

If a new version of a subject does not need to remove any of the old version's Randoop tests, please output a file anyway specifying that no tests were removed. 

You may also include a comment specifying what the file is for, that its auto-generated from this tool, that all tests in it are compileable in the old version of the program but is no longer compileable in the new version, etc.

**Review by winglam — CHANGES_REQUESTED** [review]

**Comment by winglam** [comment]
Commenting out the code instead of removing is not important. Please focus on fixing the issues mentioned above instead.

**Comment by ReedOei** [comment]
I have made the changes you requested.

**Comment by winglam** [comment]
LGTM! The change for the tool to accept multiple files is appreciated.

**Pull request merged**